### ISSUE TITLE
Crash on particular IP address

### ISSUE THREAD

**Issue by linuxhw** [secynic/ipwhois#254]
Crash of latest ipwhois 1.1 on `IP=122.215.34.1`:

```
import ipwhois

owner_info = ipwhois.IPWhois('122.215.34.1').lookup_rdap(depth=0)

print owner_info
```
Error log:

```
File "/usr/bin/XXX", line XXX, in XXX
    return ipwhois.IPWhois(ip).lookup_rdap(depth=0)
  File "/usr/lib/python2.7/site-packages/ipwhois/ipwhois.py", line 321, in lookup_rdap
    rate_limit_timeout=rate_limit_timeout
  File "/usr/lib/python2.7/site-packages/ipwhois/rdap.py", line 780, in lookup
    result_net.parse()
  File "/usr/lib/python2.7/site-packages/ipwhois/rdap.py", line 554, in parse
    self.vars[v] = self.json[v].strip()
AttributeError: 'NoneType' object has no attribute 'strip'
```

**Comment by Mekazava** [comment]
I've got this problem too. With ip 1.5.186.0 and many other from JPNIC. They have one bad field in their output
`{'rdapConformance': ['cidr0', 'rdap_level_0'], 'notices': [{'title': 'Source', 'description': ['Objects returned came from source', 'JPNIC']}, {'title': 'Terms and Conditions', 'description': ['This is the APNIC WHOIS Database query service. The objects are in RDAP format.', 'This information has been partially mirrored by APNIC from JPNIC. To obtain more specific information, please use the JPNIC WHOIS Gateway at http://www.nic.ad.jp/en/db/whois/en-gateway.html or whois.nic.ad.jp for WHOIS client. (The WHOIS client defaults to Japanese output, use the /e switch for English output)'], 'links': [{'value': 'http://jpnic.rdap.apnic.net/ip/1.5.186.0', 'rel': 'terms-of-service', 'href': 'http://www.apnic.net/db/dbcopyright.html', 'type': 'text/html'}]}], 'country': 'JP', 'events': [{'eventAction': 'last changed', 'eventDate': '2011-04-05T03:05:05Z'}], 'name': 'ODN', 'remarks': [{'description': ['SOFTBANK TELECOM Corp.'], 'title': 'description'}], 'links': [{'value': 'http://jpnic.rdap.apnic.net/ip/1.5.186.0', 'rel': 'self', 'href': 'http://jpnic.rdap.apnic.net/ip/1.5.0.0/16', 'type': 'application/rdap+json'}], 'type': None, 'endAddress': '1.5.255.255', 'ipVersion': 'v4', 'startAddress': '1.5.0.0', 'objectClassName': 'ip network', 'handle': '1.5.0.0 - 1.5.255.255', 'entities': [{'roles': ['abuse'], 'events': [{'eventAction': 'last changed', 'eventDate': '2017-10-18T10:21:54Z'}], 'links': [{'value': 'http://jpnic.rdap.apnic.net/ip/1.5.186.0', 'rel': 'self', 'href': 'http://jpnic.rdap.apnic.net/entity/IRT-JPNIC-JP', 'type': 'application/rdap+json'}], 'vcardArray': ['vcard', [['version', {}, 'text', '4.0'], ['fn', {}, 'text', 'IRT-JPNIC-JP'], ['kind', {}, 'text', 'group'], ['adr', {'label': 'Urbannet-Kanda Bldg 4F, 3-6-2 Uchi-Kanda\nChiyoda-ku, Tokyo 101-0047, Japan'}, 'text', ['', '', '', '', '', '', '']], ['email', {}, 'text', 'hostmaster@nic.ad.jp'], ['email', {'pref': '1'}, 'text', 'hostmaster@nic.ad.jp']]], 'objectClassName': 'entity', 'handle': 'IRT-JPNIC-JP'}, {'roles': ['technical', 'administrative'], 'events': [{'eventAction': 'last changed', 'eventDate': '2012-08-28T07:58:02Z'}], 'links': [{'value': 'http://jpnic.rdap.apnic.net/ip/1.5.186.0', 'rel': 'self', 'href': 'http://jpnic.rdap.apnic.net/entity/JNIC1-AP', 'type': 'application/rdap+json'}], 'vcardArray': ['vcard', [['version', {}, 'text', '4.0'], ['fn', {}, 'text', 'Japan Network Information Center'], ['kind', {}, 'text', 'group'], ['adr', {'label': 'Urbannet-Kanda Bldg 4F\n3-6-2 Uchi-Kanda\nChiyoda-ku, Tokyo 101-0047,Japan'}, 'text', ['', '', '', '', '', '', '']], ['tel', {'type': 'voice'}, 'text', '+81-3-5297-2311'], ['tel', {'type': 'fax'}, 'text', '+81-3-5297-2312'], ['email', {}, 'text', 'hostmaster@nic.ad.jp']]], 'objectClassName': 'entity', 'handle': 'JNIC1-AP'}, {'roles': ['technical', 'administrative'], 'links': [{'value': 'http://jpnic.rdap.apnic.net/ip/1.5.186.0', 'rel': 'related', 'href': 'http://whois.nic.ad.jp/cgi-bin/whois_gw?key=JP00035900/e', 'type': 'text/html', 'hreflang': 'en'}, {'value': 'http://jpnic.rdap.apnic.net/ip/1.5.186.0', 'rel': 'related', 'href': 'http://whois.nic.ad.jp/cgi-bin/whois_gw?key=JP00035900', 'type': 'text/html', 'hreflang': 'jp'}], 'objectClassName': 'entity', 'handle': 'JP00035900'}], 'cidr0_cidrs': [{'v4prefix': '1.5.0.0', 'length': 16}], 'port43': 'whois.apnic.net'}
`
`'type': None,` wich crash parsing

I make small local fix in ipwhois.rdap in _RDAPNetwork.parse
`
        for v in ['name', 'type', 'country']:

            try:

                self.vars[v] = self.json[v].strip()

-            except (KeyError, ValueError):
+            except (KeyError, ValueError, AttributeError):

                pass`

**Comment by sunny-r** [comment]
Mekazava's fix didn't work for me, instead, I check for None before applying the strip() method:

```
    for v in ['name', 'type', 'country']:

            try:
                if self.json[v] is None:
                    self.vars[v] = None
                else:
                    self.vars[v] = self.json[v].strip()
            except (KeyError, ValueError):

                pass
```

**Comment by dnk** [comment]
pull request https://github.com/secynic/ipwhois/pull/260

**Comment by secynic** [comment]
Fixed in #263: https://github.com/secynic/ipwhois/pull/263/files#diff-0a5625714b79f642cd735bee827b2f9bL556

Please test the dev branch.

**Comment by secynic** [comment]
I tested both of your IP examples, and they are working in dev. 

@sunny-r Let me know if the dev branch still isn't working for you, and I will re-open.
